### PR TITLE
Add support for default values

### LINKED ISSUE #4: Allow default values when interpolating data
It would be very useful to be able to specify a default value to fall back to if the data object doesn't contain the specified value. We could use an `||` operator to indicate the fallback. For example, with the following template:
```js
{
  "foo": "My name is {{name || 'Jane'}}"
}
```
If the data object is missing the `name` key the compilation result looks like this:
```js
{
  "foo": "My name is Jane"
}
```

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes #4

change-type: minor

### CODE REVIEW

File: doc/README.hbs [hunk 1]
```diff
@@ -71,6 +71,21 @@ The example objects contain one key and a single interpolation, but on real
 templates, there can be complex nesting levels and multiple interpolations
 (even many per property).
 
+Default values
+--------------
```
Comment (jviotti) [inline doc/README.hbs:75]: This is probably a typo, but you added two sets of dashes here :)

File: doc/README.hbs [hunk 2]
```diff
@@ -71,6 +71,21 @@ The example objects contain one key and a single interpolation, but on real
...
+--------------
+
+Default values can be provided using the `||` operator. The default value should be JSON encoded.
```
Comment (jviotti) [inline doc/README.hbs]: Ah, beautiful approach :+1:

File: doc/README.hbs [hunk 3]
```diff
@@ -71,6 +71,21 @@ The example objects contain one key and a single interpolation, but on real
...
+In the example below, if the `name` value is not provided, the default value
+`John Doe` will be used instead.
+
+```json
+{
+  "foo": "My name is \{{name || \"John Doe\"}}"
```
Comment (jviotti) [inline doc/README.hbs:83]: Should I be able to use a variable at the right of the OR operator? ie `{{name || first_name}}`? If so, this raises the question of whether we should allow any amount of chaining. For instance: `{{ name || first_name || last_name || "John Doe" }}`. What do you think?

**Review by jviotti — APPROVED** [review]
Great stuff :tada: I left an open question about chaining OR operators. I guess we can do it if its easy (in another PR), otherwise we can put it the backlog, assuming we don't have a need for it right now.

File: doc/README.hbs (hunk 3 above)
Comment (LucianBuzzo) [reply doc/README.hbs:83]: That sounds great, but I'd prefer to tackle it in a future PR! I'll create an issue so that its not forgotten.

**Comment by jviotti** [comment]
@LucianBuzzo Should we merge this, or do you want to make any chances first? (feel free to merge and release yourself if you want)

**Comment by LucianBuzzo** [comment]
@jviotti I'm waiting on the VB integration, then I'll merge and release

**Pull request merged**